### PR TITLE
PIM-5762: Removed unused filters on product datagrids

### PR DESCRIPTION
**Pull request by ClementGautier**
# Description
I removed all category filters that were wrongly injected in some datagrids only to ensure permissions related filtering in the enterprise edition.

I also added an interface for the FilterUtility because we rely on it on the enterprise edition and could make sense in the community too.

# Definition of Done

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | y
| Added Behats                      | N/A
| Changelog updated                 | y
| Review and 2 GTM                  | todo
| Micro Demo to the PO (Story only) | N/A
| Migration script                  | N/A
| Tech Doc                          | /NA

### CODE REVIEW

**Comment by juliensnz** [comment]
GTM after update the changelog

File: CHANGELOG-1.5.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+# 1.5.x
+
+## Technical improvements
+
+- PIM-5762: Removed unused category filters on product datagrids
```
Comment (juliensnz) [inline CHANGELOG-1.5.md]: add mention to the new data source filter

File: src/Pim/Bundle/FilterBundle/Filter/DatasourceFilterUtilityInterface.php [hunk 2]
```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace Pim\Bundle\FilterBundle\Filter;
+
+use Oro\Bundle\DataGridBundle\Datasource\DatasourceInterface;
+
+/**
+ * Used when you want to apply the filter from the outside, for example from a listener that access the datasource
```
Comment (phaseinducer) [inline src/Pim/Bundle/FilterBundle/Filter/DatasourceFilterUtilityInterface.php]: accesses

File: src/Pim/Bundle/FilterBundle/Filter/ProductFilterUtility.php [hunk 3]
```diff
@@ -26,4 +28,16 @@ public function applyFilter(FilterDatasourceAdapterInterface $ds, $field, $opera
     {
         $ds->getProductQueryBuilder()->addFilter($field, $operator, $value);
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function filterDatasource(DatasourceInterface $ds, $field, $operator, $value)
+    {
+        if (!$ds instanceof ProductDatasource) {
```
Comment (phaseinducer) [inline src/Pim/Bundle/FilterBundle/Filter/ProductFilterUtility.php]: Something is seriously wrong here.

File: src/Akeneo/Bundle/ClassificationBundle/Doctrine/ORM/Repository/AbstractItemCategoryRepository.php [hunk 4]
```diff
@@ -210,4 +211,24 @@ protected function getMappingConfig($item)
             'relation'           => key($categoryAssoc['relationToSourceKeyColumns']),
         ];
     }
+
+    /**
+     * @param mixed $qb
+     */
+    private function joinQueryBuilderOnCategories($qb)
```
Comment (pierallard) [inline src/Akeneo/Bundle/ClassificationBundle/Doctrine/ORM/Repository/AbstractItemCategoryRepository.php]: protected please

**Pull request merged**